webapp/passport/js/peakareachart.js

```diff
@@ -0,0 +1,357 @@
+            .attr("width", relativeWidth)
+            .attr("y", yRelative(46))
+            .attr("height", 16);
+
+
+        protein_domain.append("text")
+            .attr("class", "highlight_peptide_bounds_text")
+            .style("text-anchor", "middle")
+            .attr("x", relativeStart + relativeWidth/2)
+            .attr("y", yRelative(31 - 12 - 8)) //12px font  8px for highlight_peptide_bounds
+            .text(start +  " - " + (start + sequence.length -1)); // subtract 1 because end index will be base 2
+
+    },
+    paintPeptide: function paintPeptide(d) {
+        var parentWidth = parseInt(d3.select('#chart').style('width'), 10);
+        var sequence = d.Sequence;
+        var peptideWidth = sequence.length * 50;
+        var marginLeft = (parentWidth - peptideWidth)/2;
+        if(d3.select("#peptide")[0][0].getAttribute("peptide") != sequence) {
+            d3.select("#peptide").selectAll("svg").remove();
+            var svg =  d3.select("#peptide").append("svg")
+                .attr("width", peptideWidth)
+                .attr("height", 50)
+                .attr("peptide", sequence)
+                .attr("style", "margin-left:"+marginLeft+"px;");
+
+            svg[0][0].innerHTML = "";
+            svg[0][0].setAttribute("peptide", sequence);
+            for(var i = 0; i < sequence.length; i++) {
+                // issues with i = 0 because 'cx' would be 0*50 and be off the page, so start i at 1
+                svg.append("circle")
+                    .attr("r", 25)
+                    .attr("cx", 25 + (i* 50))
+                    .attr("cy", 25)
+                    .attr("class", "aminoCircle");
+
+                svg.append("text")
+                    .attr("x", 25+ (i* 50))
+                    .attr("y", 32) // TODO figure out resizing
+                    .attr("text-anchor", "middle")
+                    .attr("style","font:20px 'Raleway', sans-serif; font-weight:600;")
+                    .text(sequence[i]);
+            }
+        }
+    },
+    selectPeptide: function selectPeptide(p) {
+        if(p == null)
+            return;
+        // order is VERY important here so be careful and make sure changes don't break anything
+        var oldPeptide = protein.oldPeptide;
+        var selectedPeptide = protein.selectedPeptide;
+        if(oldPeptide != null) {
+            $("#group-"+ oldPeptide.Sequence).find(".left-bar").css("fill","#8a89a6");
+            $("#group-"+ oldPeptide.Sequence).find(".right-bar").css("fill","#a05d56");
+        }
+
+        $("#group-"+ selectedPeptide.Sequence).find(".left-bar").css("fill","#494777");
+        $("#group-"+ selectedPeptide.Sequence).find(".right-bar").css("fill","#66221B");
+
+
+        // highlights peptide in left panel and soon also protein sequence
+        if(oldPeptide == null || oldPeptide != p.Sequence) {
+            if(oldPeptide != null) {
+                $('.'+oldPeptide.Sequence+'-text').css("background-color","transparent");
```

Comment:
Are there any encoding concerns here?

Reply:
this line is getting the element and updating the css. I don't think it needs to be encoded as this won't be displayed/executed on the page.